trino-lb-core/src/trino_api/trino_events.rs

```diff
@@ -0,0 +1,69 @@
+use std::fmt::Display;
+
+use serde::{Deserialize, Serialize};
+use url::Url;
+
+use crate::TrinoQueryId;
+
+#[derive(Clone, Debug, PartialEq, Eq, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct TrinoEvent {
+    pub metadata: TrinoEventMetadata,
+}
+
+impl TrinoEvent {
+    pub fn query_state(&self) -> &TrinoQueryState {
+        &self.metadata.query_state
+    }
+
+    pub fn query_id(&self) -> &TrinoQueryId {
+        &self.metadata.query_id
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct TrinoEventMetadata {
+    pub uri: Url,
+    pub query_id: TrinoQueryId,
+    pub query_state: TrinoQueryState,
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Deserialize, Serialize)]
+#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
+pub enum TrinoQueryState {
+    Queued,
+    Executing,
+    Finished,
+}
+
+impl Display for TrinoQueryState {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            TrinoQueryState::Queued => write!(f, "queued"),
+            TrinoQueryState::Executing => write!(f, "executing"),
+            TrinoQueryState::Finished => write!(f, "finished"),
```

Comment:
question: Do we want the display string to use the same casing which is used during ser/de to be consistent to Trino itself?

Reply:
No strong opinion, 0b3a6a0

Reply:
We use strum in other places, to configure casing and also reduce typos.